main.py

```diff
@@ -386,68 +210,36 @@ def main():
-            pod_state_expire_duration=args.pod_expire_duration,
-            pod_without_metadata_state_expire_duration=args.pod_without_metadata_expire_duration,
-        ),
-        publisher=publisher,
+    message_handler = MessageHandler(
+        system_meta={
+            "node": args.waggle_node_id,
+            "vsn": args.waggle_node_vsn,
+        },
     )
 
-    while True:
-        with ExitStack() as es:
-            logging.info(
-                "connecting consumer to rabbitmq server at %s:%d as %s.",
-                params.host,
-                params.port,
-                params.credentials.username,
-            )
-
-            connection = es.enter_context(pika.BlockingConnection(params))
-            channel = es.enter_context(connection.channel())
-
-            logging.info("setting up queues and exchanges.")
-            channel.exchange_declare("data.topic", exchange_type="topic", durable=True)
-            declare_exchange_with_queue(channel, "to-validator")
-            declare_exchange_with_queue(channel, "to-beehive")
-
-            pod_event_watcher = es.enter_context(PluginPodEventWatcher())
-
-            def forward_pod_events():
-                logging.debug("updating pod events...")
-                for pod in pod_event_watcher.ready_events():
-                    if pod is None:
-                        raise RuntimeError("pod watcher terminated")
-                    handler.handle_pod(pod)
-                logging.debug("updated pod events")
-
-            call_every(connection, 1.0, forward_pod_events)
-            call_every(connection, 10.0, handler.handle_expired_pods)
-            amqp.consume(channel, "to-validator", handler.handle_delivery)
-
-            logging.info("starting consumer...")
-            try:
-                channel.start_consuming()
-            except KeyboardInterrupt:
-                return
-            except Exception:
-                logging.exception("consumer disconnected. will retry...")
-
-        # # TODO(sean) figure out clean way of handling this. when the channel expires, I assume the
-        # outstanding deliveries must be consumed again.
-        logging.info("clearing backlogged messages but keeping pod state")
-        handler.clear_backlogs_but_keep_pod_state()
-        time.sleep(3.0)
-
-        # we can just allow the cache to expire stuff separately...
+    with ExitStack() as es:
+        logging.info("connecting consumer to rabbitmq server at %s:%d as %s.",
+            params.host,
+            params.port,
+            params.credentials.username,
+        )
+
+        connection = es.enter_context(pika.BlockingConnection(params))
+        channel = es.enter_context(connection.channel())
+
+        logging.info("setting up queues and exchanges.")
+        channel.exchange_declare("data.topic", exchange_type="topic", durable=True)
+        declare_exchange_with_queue(channel, "to-validator")
```

Comment:
"to-validator" and "to-beehive" variables?

Reply:
Done. Added args for src / dst queues and exchanges.